lib/Alchemy/Phrasea/Databox/DataboxRepository.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/*
+ * This file is part of Phraseanet
+ *
+ * (c) 2005-2015 Alchemy
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Alchemy\Phrasea\Databox;
+
+use Alchemy\Phrasea\Application;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class DataboxRepository implements DataboxRepositoryInterface
+{
+    /** @var Application */
+    private $app;
+    /** @var \appbox */
+    private $appbox;
+    /** @var DataboxFactory */
+    private $factory;
+
+    public function __construct(Application $app, \appbox $appbox, DataboxFactory $factory = null)
+    {
+        $this->app = $app;
+        $this->appbox = $appbox;
+        $this->factory = $factory ?: new DataboxFactory();
+    }
+
+    /**
+     * @param int $id
+     * @return \databox|null
+     */
+    public function find($id)
+    {
+        try {
+            $databox = $this->factory->create($this->app, (int)$id);
+        } catch (NotFoundHttpException $exception) {
```

Comment:
You should explain with a small comment why there is this weird exception catching here.